src/Services/Endpoint.php

```diff
@@ -42,19 +44,31 @@ class Endpoint extends SolariumEndpoint {
    * @param array $options
    *   Array of options for the endpoint. Currently,
    *   they are used by other functions of the endpoint.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
    */
   public function __construct(array $options = []) {
-    // We intentionally want to override this options in case they are set in the parameter.
-    $options = array_merge($options, [
-      'scheme' => getenv('PANTHEON_INDEX_SCHEME'),
-      'host' => getenv('PANTHEON_INDEX_HOST'),
-      'port' => getenv('PANTHEON_INDEX_PORT'),
-      'path' => getenv('PANTHEON_INDEX_PATH'),
-      'core' => getenv('PANTHEON_INDEX_CORE'),
-      'schema' => getenv('PANTHEON_INDEX_SCHEMA'),
-      'collection' => NULL,
-      'leader' => FALSE,
-    ]);
+    $storage = \Drupal::entityTypeManager()->getStorage('search_api_server');
+    /** @var \Drupal\search_api\ServerInterface $server */
+    $server = $storage->load(self::DEFAULT_NAME);
```

Comment:
I'm not sure what's the right approach, but I'd like to raise a concern about this line forcing users to create the server with a "hardcoded" machine name. This is not common practice in other Solr providers and could lead to lots of problems from customers.

What if they screw up the server that was created during the module installation and then decide to create a new one and the name is different?

What if they're updating from old search_api_pantheon and then the config is not installed and their server is named different?

Reply:
My understanding is that `search_api_pantheon` module must provide a single predefined server configuration for the platform, so it fits this hardcoded approach we currently have well. 
If this is the case, my main concern would be that we missing the following features:
* prevent a user from deleting the server config (ie. "delete" context action)
* prevent a user from creating an additional `Pantheon Search Connector`-based server config (ie. "Add Server")

@stovak what are your thoughts on this?